Release VisionFive 2 Changes

NebraskaCoder opened the pull request:
This pull request introduces support for VisionFive 2 images and refactors the handling of Raspberry Pi images into a more generic structure for specialized devices. It also updates the UI and translations to accommodate these changes. Below are the key updates grouped by theme:

### Feature Additions
* Added support for VisionFive 2 images, including download links and checksums, in `data/downloads.json`. [[1]](diffhunk://#diff-aaf3ac0660fa4e2e48faac71aa7afe2816671df41544074bc7ef178dc5e397d9R442-R444) [[2]](diffhunk://#diff-aaf3ac0660fa4e2e48faac71aa7afe2816671df41544074bc7ef178dc5e397d9R456-R458)
* Introduced a new `SpecializedDevice` interface and updated `DownloadOptions` and `Links` interfaces in `types/downloads.ts` to handle specialized devices like Raspberry Pi. [[1]](diffhunk://#diff-9b4e10b44a1565c659edd6ed75b681f03e8c624493101a43808861ed795d2368R20-R26) [[2]](diffhunk://#diff-9b4e10b44a1565c659edd6ed75b681f03e8c624493101a43808861ed795d2368R54-R57) [[3]](diffhunk://#diff-9b4e10b44a1565c659edd6ed75b681f03e8c624493101a43808861ed795d2368R81-R89)

### UI Enhancements
* Added a new `DefaultImageCard` component for VisionFive 2 images in the `DownloadPage` component, with a mapping function to format version data. ([app/[locale]/download/page.tsxR176-R208](diffhunk://#diff-0147ff569755f73a418a60fe152d6428b369352aad4d74c4f026146204c9a7c5R176-R208), [app/[locale]/download/page.tsxR308-R318](diffhunk://#diff-0147ff569755f73a418a60fe152d6428b369352aad4d74c4f026146204c9a7c5R308-R318))

### Data Structure Updates
* Refactored Raspberry Pi image data in `data/downloads.json` into a `specializedDevices` array for better extensibility. [[1]](diffhunk://#diff-aaf3ac0660fa4e2e48faac71aa7afe2816671df41544074bc7ef178dc5e397d9L167-R174) [[2]](diffhunk://#diff-aaf3ac0660fa4e2e48faac71aa7afe2816671df41544074bc7ef178dc5e397d9L187-L190) [[3]](diffhunk://#diff-aaf3ac0660fa4e2e48faac71aa7afe2816671df41544074bc7ef178dc5e397d9L223-R231) [[4]](diffhunk://#diff-aaf3ac0660fa4e2e48faac71aa7afe2816671df41544074bc7ef178dc5e397d9L243-L246) [[5]](diffhunk://#diff-aaf3ac0660fa4e2e48faac71aa7afe2816671df41544074bc7ef178dc5e397d9L270-L272)

### Localization Updates
* Added translations for VisionFive 2 and specialized devices in `messages/en.json`. [[1]](diffhunk://#diff-5d9e5048516ec2fe83ca06813e79c3b8b44c6c96294f25c7311db311e28eeaeeR261-R270) [[2]](diffhunk://#diff-5d9e5048516ec2fe83ca06813e79c3b8b44c6c96294f25c7311db311e28eeaeeL270-R280)

hbjydev commented:
@domingoruiz See the build failure

domingoruiz commented:
And now @ressonix ?

hbjydev commented on vercel.json:
```diff
@@ -30,6 +30,7 @@
     { "src": "/ca/(.*)", "status": 404, "dest": "/ca/404.html" },
-    { "src": "/(.*)",    "status": 404, "dest": "/404.html" }
+    { "src": "/(.*)",    "status": 404, "dest": "/404.html" },
+    { "src": "/(.*)/posts/", "status": 200, "dest": "/posts/" }
```
Couple things:
1. this needs to be before the global 404 route
2. this file routes URLs to files, not URLs to URLs, so this solution wouldn't work unfortunately.

domingoruiz replied:
That was the error, instead of a 200 it is a 404. Changing it like this should work. Although if we are going to change to a static site we will not have this problem.

hbjydev commented:
Sorry Domingo, but on reviewing the PR, I realise this solution won't work for Vercel.

And once we switch to StaticDeploy, this issue will go away anyway.

NebraskaCoder committed:
Add VisionFive 2 images support to download page

- Introduced mapping for VisionFive 2 version items in the download page.
- Updated download options and links for VisionFive 2 images in the JSON data.
- Added localization strings for VisionFive 2 images in English messages.
- Extended TypeScript interfaces to include VisionFive 2 download options and links.

NebraskaCoder committed:
Update DownloadPage to conditionally display readMe link for VisionFive 2 images

- Made the readMe link optional in the TypeScript interface for VisionFive 2 images.
- Updated the DownloadPage component to only include the readMe link if it exists, improving the handling of undefined links.

NebraskaCoder committed:
Merge branch 'develop' into feature/riscv-specialized-hardware-section

NebraskaCoder committed:
Refactor downloads.json to introduce specializedDevices section for Raspberry Pi and VisionFive 2 images, enhancing organization and clarity. Update types and messages to support new structure.

NebraskaCoder committed:
Merge branch 'develop' into feature/riscv-specialized-hardware-section

NebraskaCoder committed:
Enhance downloads.json by adding VisionFive 2 image download and checksum links, improving resource accessibility for specialized devices.

NebraskaCoder committed:
Refactor downloads.json to consolidate VisionFive 2 image links, removing redundant entries for improved clarity and accessibility.

NebraskaCoder committed:
Update VisionFive 2 image download link in downloads.json to reflect the latest file format, ensuring accurate resource access.

NebraskaCoder committed:
Merge pull request #186 from rocky-linux/feature/riscv-specialized-hardware-section

Support for VisionFive 2 images and Update structure for specialized device downloads
